server/src/main/java/org/elasticsearch/cluster/service/ClusterApplierService.java

```diff
@@ -527,8 +543,11 @@ private void callClusterStateListener(ClusterChangedEvent clusterChangedEvent, S
         for (ClusterStateListener listener : listeners) {
             try {
                 logger.trace("calling [{}] with change to version [{}]", listener, clusterChangedEvent.state().version());
-                try (Releasable ignored = stopWatch.timing("notifying listener [" + listener + "]")) {
+                final String name = listener.toString();
+                try (Releasable ignored = stopWatch.timing("notifying listener [" + name + "]")) {
```

Comment:
Could we stop using the stopwatch altogether and move the timing into the new tracker? This'd also let us track the other things that aren't listeners/appliers (e.g. connecting to new nodes).

Reply:
Yes, that makes sense. But it does mean that slow appliers/listeners will not show up in the slow cluster state applier warning log. I think that is ok? Since slow appliers/listeners are now visible via the node stats api.